### PR TITLE
raise missing result on server side

### PR DESCRIPTION
**Pull request by damnever**
Other thrift client may has different behavior from thriftpy.

@wooparadog

### CODE REVIEW

**Comment by damnever** [comment]
Tornado4.4 removed support for Python 2.6.

File: thriftpy/thrift.py [hunk 1]
```diff
@@ -229,10 +229,6 @@ def _recv(self, _api):
             if k != "success" and v:
                 raise v
 
-        # no throws & not void api
```
Comment (wooparadog) [inline thriftpy/thrift.py]: We maybe should not modify logics in client side ?

File: thriftpy/thrift.py [hunk 2]
```diff
@@ -297,13 +293,23 @@ def process(self, iprot, oprot):
...
             return self.send_exception(oprot, api, result, seqid)
 
+        has_exc = False
         try:
-            result.success = call()
+            success = call()
         except Exception as e:
             # raise if api don't have throws
             self.handle_exception(e, result)
+            has_exc = True
```
Comment (wooparadog) [inline thriftpy/thrift.py:306]: It makes sense that  one-way calls don't even need to return an exception. But what's the behavior in official implementation?

File: thriftpy/thrift.py [hunk 3]
```diff
@@ -297,13 +293,23 @@ def process(self, iprot, oprot):
...
 
         if not result.oneway:
+            # raise missing result on server side
+            if not has_exc and hasattr(result, 'success'):
+                if success is None:
+                    raise TApplicationException(
```
Comment (wooparadog) [inline thriftpy/thrift.py:312]: For void methoda, return is actually empty ?

File: thriftpy/thrift.py (hunk 2 above)
Comment (damnever) [reply thriftpy/thrift.py:306]: > null return - null cannot be returned directly from a function. Use a wrapper struct or a marker value instead

**Pull request closed without merging**